File: Food-Journal-Backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace Food_Journal.Models
{
    public class Product
    {
        [Required]
        public Guid Id { get; init; }
        [Required]
        public Guid? UserId { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public double Proteins { get; set; }
        [Required]
        public double Fats { get; set; }
        [Required]
        public double Carbohydrates { get; set; }
        [Required]
        public double Kcal { get; set; }

    }
}

File: Food-Journal-Frontend/src/app/interfaces/product.ts
export interface IProduct {
    id: string,
    name: string,
    proteins: number,
    fats: number,
    carbohydrates: number,
    kcal: number   
}

export interface IProductRequest {
    name: string,
    proteins: number,
    fats: number,
    carbohydrates: number,
    // kcal: number   
}

export interface IAddProduct {
    name: string,
    proteins: number,
    fats: number,
    carbohydrates: number,
}